app/src/main/java/ch/epfl/polybazaar/listing/ListingDatabase.java

```diff
@@ -30,7 +34,8 @@
      */
     public static void fetchListing(final String listingID, final ListingCallback callback){
         final ListingCallbackAdapter adapterCallback = new ListingCallbackAdapter(callback);
-        fetchData(listingCollectionName, listingID, adapterCallback);
+        db = DatastoreFactory.getDependency();
+        db.fetchData(listingCollectionName, listingID, adapterCallback);

```

Comment:
The factory's dependecy is static, if you make all classes implementing DataStore abstract, you can lose the db. and assignment. That will be much easier to use and makes more sense, since we won't be using the MockDatastore and the FirebaseDatastore at the same time.